--- a/Tools/DumpRenderTree/qt/DumpRenderTreeMain.cpp
+++ b/Tools/DumpRenderTree/qt/DumpRenderTreeMain.cpp
@@ -115,7 +115,9 @@
     if (suppressQtDebugOutput)
         qInstallMessageHandler(messageHandler);
 
+/*
     WebKit::QtTestSupport::initializeTestFonts();
+*/
 
     QApplication::setStyle(QStyleFactory::create(QLatin1String("windows")));
     QApplication::setDesktopSettingsAware(false);
--- a/Tools/DumpRenderTree/qt/DumpRenderTreeQt.cpp
--- b/Tools/DumpRenderTree/qt/DumpRenderTreeQt.cpp
@@ -622,10 +622,10 @@
     QFocusEvent ev(QEvent::FocusIn);
     m_page->event(&ev);
 
+/*
     WebKit::QtTestSupport::clearMemoryCaches();
-
     WebKit::QtTestSupport::initializeTestFonts();
-
+*/
     DumpRenderTreeSupportQt::dumpFrameLoader(url.toString().contains("loading/"));
     setTextOutputEnabled(true);
     m_page->mainFrame()->load(url);
