--- a/Source/WebKit/qt/Api/qwebsettings.h
+++ b/Source/WebKit/qt/Api/qwebsettings.h
@@ -87,7 +87,9 @@
         CaretBrowsingEnabled,
         NotificationsEnabled,
         WebAudioEnabled,
-        CanvasPathEnabled
+        CanvasPathEnabled,
+        RepaintCounter,
+        DebugBorder
     };
     enum WebGraphic {
         MissingImageGraphic,

--- a/Source/WebKit/qt/Api/qwebsettings.cpp
+++ b/Source/WebKit/qt/Api/qwebsettings.cpp
@@ -153,8 +153,16 @@
         settings->setAcceleratedCompositingEnabled(value);
 
         bool showDebugVisuals = qgetenv("WEBKIT_SHOW_COMPOSITING_DEBUG_VISUALS") == "1";
-        settings->setShowDebugBorders(showDebugVisuals);
-        settings->setShowRepaintCounter(showDebugVisuals);
+
+        value = attributes.value(QWebSettings::DebugBorder,
+                                      global->attributes.value(QWebSettings::DebugBorder));
+
+        settings->setShowDebugBorders(showDebugVisuals || value);
+
+        value = attributes.value(QWebSettings::RepaintCounter,
+                                      global->attributes.value(QWebSettings::RepaintCounter));
+
+        settings->setShowRepaintCounter(showDebugVisuals || value);
 #endif
 #if ENABLE(WEBGL)
         value = attributes.value(QWebSettings::WebGLEnabled,
@@ -566,6 +574,9 @@
     d->attributes.insert(QWebSettings::CaretBrowsingEnabled, false);
     d->attributes.insert(QWebSettings::NotificationsEnabled, true);
     d->attributes.insert(QWebSettings::CanvasPathEnabled, false);
+    d->attributes.insert(QWebSettings::RepaintCounter, false);
+    d->attributes.insert(QWebSettings::DebugBorder, false);
+
     d->offlineStorageDefaultQuota = 5 * 1024 * 1024;
     d->defaultTextEncoding = QLatin1String("iso-8859-1");
     d->thirdPartyCookiePolicy = AlwaysAllowThirdPartyCookies;
