--- a/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp
+++ b/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp
@@ -565,6 +565,14 @@
             }
         }
 
+       if (antialias) {
+            // If antialiasing is not fully disabled, fontconfig may still disable it on a font match basis.
+            FcBool fc_antialias;
+            if (FcPatternGetBool(match, FC_ANTIALIAS,0, &fc_antialias) != FcResultMatch)
+                fc_antialias = true;
+            antialias = fc_antialias;
+        }
+
         if (f.hintingPreference == QFont::PreferDefaultHinting) {
             const QPlatformServices *services = QGuiApplicationPrivate::platformIntegration()->services();
             if (services && (services->desktopEnvironment() == "GNOME" || services->desktopEnvironment() == "UNITY")) {
