--- a/mkspecs/devices/linux-mipsel-broadcom-97425-g++/qeglfshooks_bcm.cpp
+++ b/mkspecs/devices/linux-mipsel-broadcom-97425-g++/qeglfshooks_bcm.cpp
@@ -55,6 +55,10 @@
 #include "nexus_display.h"
 #include "nexus_core_utils.h"
 
+#if defined(NEXUS_SERVER_SUPPORT)
+#include "refsw_session_simple_client.h"
+#endif
+
 #include "default_nexus.h"
 
 #define UNUSED(X) (void)X
@@ -70,18 +74,33 @@
 static unsigned int gs_screen_wdt   = 0;
 static unsigned int gs_screen_hgt   = 0;
 
-static NEXUS_DisplayHandle  gs_nexus_display = 0;
 static void                 *gs_native_window = 0;
 
+#if !defined(NEXUS_SERVER_SUPPORT)
+static NEXUS_DisplayHandle  gs_nexus_display = 0;
 static NXPL_PlatformHandle  nxpl_handle = 0;
+#endif 
 
 bool InitPlatform(void)
 {
-    NEXUS_PlatformSettings        platform_settings;
     NEXUS_Error                   err;
 
+#if defined(NEXUS_SERVER_SUPPORT)
+
+    NEXUS_ClientAuthenticationSettings authSettings;
+
+    qWarning("Calling NEXUS_Platform_AuthenticatedJoin");
+    simple_client_init("xre", &authSettings);
+
+    err = NEXUS_Platform_AuthenticatedJoin(&authSettings);
+
+#else
+
+    NEXUS_PlatformSettings        platform_settings;
+
     /* Initialise the Nexus platform */
     NEXUS_Platform_GetDefaultSettings(&platform_settings);
+
 #if NEXUS_DTV_PLATFORM
 #ifdef SIXTY_HZ
     platform_settings.displayModuleSettings.panel.dvoLinkMode = NEXUS_PanelOutputLinkMode_eDualChannel1;
@@ -91,6 +110,9 @@
     platform_settings.openFrontend = false;
 
     err = NEXUS_Platform_Init(&platform_settings);
+
+#endif
+
     if (err)
     {
         qFatal("Err: NEXUS_Platform_Init() failed");
@@ -100,6 +122,7 @@
     return true;
 }
 
+#if !defined(NEXUS_SERVER_SUPPORT)
 
 NEXUS_DisplayHandle OpenDisplay(NEXUS_VideoFormat format)
 {
@@ -245,13 +268,18 @@
 #endif
 }
 
+#endif 
+
 bool InitDisplay(void)
 {
 
     NXPL_NativeWindowInfo   win_info;
+
+#if !defined(NEXUS_SERVER_SUPPORT)
     NEXUS_GraphicsSettings  graphics_settings;
     NEXUS_DisplaySettings   disp_settings;
     NEXUS_VideoFormatInfo   video_format_info;
+#endif
 
     if (InitPlatform() == false)
     {
@@ -259,6 +287,13 @@
         return false;
     }
 
+#if defined(NEXUS_SERVER_SUPPORT)
+
+    gs_screen_wdt = 1280;
+    gs_screen_hgt = 720; 
+
+#else
+
     gs_nexus_display = OpenDisplay((NEXUS_VideoFormat)0);
     if (gs_nexus_display == NULL)
     {
@@ -283,6 +318,8 @@
 
     NEXUS_Display_GetGraphicsSettings(gs_nexus_display, &graphics_settings);
 
+#endif
+
     win_info.x = 0; 
     win_info.y = 0;
     win_info.width = gs_screen_wdt;
@@ -300,6 +337,9 @@
 
 void DeInitPlatform(void)
 {
+
+#if !defined(NEXUS_SERVER_SUPPORT)
+
     if (gs_nexus_display != 0)
     {
         NXPL_DestroyNativeWindow(gs_native_window);
@@ -311,6 +351,9 @@
 
     /* Close the platform */
     NEXUS_Platform_Uninit();
+
+#endif
+
 }
 
 #ifdef __cplusplus
