--- a/mkspecs/devices/linux-rasp-pi-g++/qmake.conf
+++ b/mkspecs/devices/linux-rasp-pi-g++/qmake.conf
@@ -6,17 +6,19 @@
 
 QT_QPA_DEFAULT_PLATFORM = wayland
 
-QMAKE_LFLAGS           += -Wl,-rpath-link,$$[QT_SYSROOT]/opt/vc/lib
+QMAKE_LFLAGS            += -Wl,-rpath-link,$$[QT_SYSROOT]/usr/lib
 
-QMAKE_LIBDIR_OPENGL_ES2 = $$[QT_SYSROOT]/opt/vc/lib
-QMAKE_LIBDIR_EGL        = $$QMAKE_LIBDIR_OPENGL_ES2
+QMAKE_LIBDIR_EGL        = $$[QT_SYSROOT]/usr/lib
+QMAKE_LIBDIR_OPENGL_ES2 = $$[QT_SYSROOT]/usr/lib
+QMAKE_LIBDIR_OPENVG     = $$[QT_SYSROOT]/usr/lib
 
-QMAKE_INCDIR_EGL        = $$[QT_SYSROOT]/opt/vc/include \
-                          $$[QT_SYSROOT]/opt/vc/include/interface/vcos/pthreads \
-                          $$[QT_SYSROOT]/opt/vc/include/interface/vmcs_host/linux
+QMAKE_INCDIR_EGL        = $$[QT_SYSROOT]/usr/include $$[QT_SYSROOT]/usr/include/interface/vcos/pthreads $$[QT_SYSROOT]/usr/include/interface/vmcs_host/linux
 QMAKE_INCDIR_OPENGL_ES2 = $${QMAKE_INCDIR_EGL}
 
 QMAKE_LIBS_EGL          = -lEGL -lGLESv2
+QMAKE_INCDIR_OPENVG     = $$[QT_SYSROOT]/usr/lib
+QMAKE_LIBS_OPENVG       = -lOpenVG
+QMAKE_LIBS_OPENVG      += $${QMAKE_LIBS_EGL}
 
 contains(DISTRO, squeeze) {
     #Debian Squeeze: Legacy everything

