--- a/examples/egl/testegl.c
+++ b/examples/egl/testegl.c
@@ -870,7 +870,8 @@
     gpointer user_data)
 {
   APP_STATE_T *state = (APP_STATE_T *) user_data;
-  queue_object (state, GST_MINI_OBJECT_CAST (gst_buffer_ref (buffer)), TRUE);
+//  queue_object (state, GST_MINI_OBJECT_CAST (gst_buffer_ref (buffer)), TRUE);
+  queue_object (state, GST_MINI_OBJECT_CAST (gst_buffer_ref (buffer)), FALSE);
 }
 
 static gboolean
